cns/restserver/util.go

```diff
@@ -236,37 +237,60 @@ func (service *HTTPRestService) updateIpConfigsStateUntransacted(req cns.CreateN
 	}
 
 	// Add the newIpConfigs, ignore if ip state is already in the map
-	service.addIPConfigStateUntransacted(req.NetworkContainerid, newIPConfigs)
+	service.addIPConfigStateUntransacted(req.NCVersion, req.NetworkContainerid, newIPConfigs)
 
 	return 0, ""
 }
 
 // addIPConfigStateUntransacted adds the IPConfigs to the PodIpConfigState map with Available state
 // If the IP is already added then it will be an idempotent call. Also note, caller will
 // acquire/release the service lock.
-func (service *HTTPRestService) addIPConfigStateUntransacted(ncId string, ipconfigs map[string]cns.SecondaryIPConfig) {
-	// add ipconfigs to state
-	for ipId, ipconfig := range ipconfigs {
-		// if this IPConfig already exists in the map, then ignore as this is an idempotent state
-		if _, exists := service.PodIPConfigState[ipId]; exists {
-			continue
-		}
+func (service *HTTPRestService) addIPConfigStateUntransacted(ncVersion, ncId string, ipconfigs map[string]cns.SecondaryIPConfig) {
```

Comment:
I think in this function you also need to pass existingIpconfigs so that you can keep those IPs as available and mark the new ones in PendingProgramming State

Reply:
Yes, existingSecondaryIPConfigs is needed.